### DISCUSSION TITLE
How to center `outline` title?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2004] [Q&A] [answered]
I've tried fiddling with `#set outline` and `#show outline`, but I don't understand how to (if possible) center it (`align(center)`). Can someone please help.

I'm thinking that the workaround can be the same as in LaTeX (bibliography titles etc.): first write `align(center, title)` and then write `outline(title: none)`.

Since I mostly need uniform titles everywhere, I can just write `#show heading: align.with(center)` or `#show heading: it => align(center, it)` and it will center every `heading` including `outline`'s title.

**Comment by xkevio** [comment]
`#show outline: align.with(center)` would technically center the whole outline but unless you have a custom one, it won't mess with the body.

EDIT: Actually, nvm, you can just do:
```
#show outline: it => {
  show heading: set align(center)
  it
}
```

**Reply by Andrew15-5** [reply]
Good workaround. Interesting if there is possibility now or in the future for it to affect the body with the title of the `outline`. There is really should be something like `#set outline(title: it => align(center, it))` or `#outline(title: align(center)[title])`

**Reply by laurmaedje** [reply]
It is not really a workaround. It is the intended way and way more general that making the title parameter take a function.

**Reply by Andrew15-5** [reply]
@laurmaedje, if it is the intended way, then the logical structure/syntax is wrong, because it says, "if you see an `outline`, then center it whole". But the user wants to only center the title of the outline. Typst is supposed to be very easy to understand and be readable (with comparison to LaTeX), if I understand correctly. The closer the "code" part to natural language/logic, the better. Anyway, the edit is good and there is the same answer here: https://github.com/typst/typst/discussions/2004#discussioncomment-6825197

**Reply by PgBiel** [reply]
> if it is the intended way, then the logical structure/syntax is wrong, because it says, "if you see an `outline`, then center it whole". 

I think there was a misunderstanding here; Laurenz seems to have referred to @xkevio's edit, which includes the proper answer:


> EDIT: Actually, nvm, you can just do:
> 
> ```
> #show outline: it => {
>   show heading: set align(center)
>   it
> }
> ```

**Reply by Andrew15-5** [reply]
If so, then it's a misunderstanding.

**Comment by PgBiel** [comment, marked as answer]
maybe 
```
#show outline: it => {
  show heading: set align(center)
  it
}
```
?

**Reply by xkevio** [reply]
Yea I just wanted to reply with that. This is the better and intended way.

**Reply by Andrew15-5** [reply]
Oh, yeahhhh, hmm. I was trying to do the same thing: `#show outline: show heading: align(center)`, but I can't use nested `show` rules....that is, unless you nest the nested rule inside a function!

<details><summary>meme</summary>

![meme](https://github.com/typst/typst/assets/37143421/9feca197-be10-4377-b5f1-fcf9dc014c28)

[meme-source.zip](https://github.com/typst/typst/files/12442250/meme-source.zip)

</details>

I don't think it will change much, but you can also do this:

```typ
#show outline: it => { show heading: align.with(center);it }
#show outline: it => { show heading: title => align(center, title);it }
```

**Reply by PgBiel** [reply]
We could definitely add a hint suggesting this... or support nested show rules :p

**Reply by Andrew15-5** [reply]
Nested show rules sounds like conditional hell in coding, but I don't think you can abuse it like `if-else` statement, so it should be all good.

**Reply by laurmaedje** [reply]
Nested show rule would be very easy to support. I intentionally forbade it at the time being because it reads weirdly. Ideally the nesting could be expressed with a single composite selector.

**Reply by Andrew15-5** [reply]
Almost like CSS selector!))

**Reply by laurmaedje** [reply]
Yes!

**Reply by PgBiel** [reply]
Maybe the fabled `in` selector could just become syntactic sugar for nested show rules instead of its own complicated thing? (Of course, that might not be the most efficient approach, however.)

**Reply by laurmaedje** [reply]
Perhaps.
